README.md

```diff
@@ -89,41 +90,64 @@ That's it! You have authenticated the user using Auth0.
 
-You can change the authentication behavior to use **Implicit Grant** instead. To do this you'll need to check in your Applications's Settings on the [Dashboard](https://manage.auth0.com/#/applications) which Algorithm is used by the Server to sign the tokens. The default algorithm is `HS256`, but it can be changed to `RS256` in the "Advanced Settings" section on the "OAuth" tab. Below you'll find some configuration examples:
+#### Signing Algorithms
 
+The **HS256 symmetric algorithm** is the default expected signing algorithm. Tokens are signed and verified using the client secret found in your Auth0 Application's settings. You use this value when you instantiate the `AuthenticationController` instance.
 
-#### Using Implicit Grant with HS256 algorithm
+If your application is using the **RS256 asymmetric algorithm**, tokens are signed using a private key and verified using the public key associated with your Auth0 domain.
+If using RS256, configure a `JwkProvider` for your Auth0 domain to enable retrieving the public key needed during the verification phase: 
 
-The token's are signed by the Auth0 Server using the `Client Secret`.
 
 ```java
+JwkProvider jwkProvider = new JwkProviderBuilder("domain").build();
 AuthenticationController authController = AuthenticationController.newBuilder("domain", "clientId", "clientSecret")
-    .withResponseType("id_token")
+    .withJwkProvider(jwkProvider)
     .build();
 ```
 
-#### Using Implicit Grant with RS256 algorithm.
+The `JwkProvider` returned from the `JwkProviderBuilder` is cached and rate limited by default. Please see the [jwks-rsa-java repository](https://github.com/auth0/jwks-rsa-java) to learn how to customize these options.
 
-The tokens are signed using the Private Key. To verify them, the **Public Key** certificate must be obtained from a trusted source like the `well-known.json` file, which can be located locally or hosted by a server. For this example, we will use the one Auth0 hosts for your application. We can obtain it using the application's domain:
+#### OAuth Flows
 
+The [Authorization Code Flow](https://auth0.com/docs/flows/concepts/auth-code) is the default authorization flow.
+
+To use the [Implicit Grant Flow](https://auth0.com/docs/flows/concepts/implicit), configure the `AuthenticationController` with the `id_token` response type:
 
 ```java
-JwkProvider jwkProvider = new JwkProviderBuilder("domain").build();
 AuthenticationController authController = AuthenticationController.newBuilder("domain", "clientId", "clientSecret")
     .withResponseType("id_token")
-    .withJwkProvider(jwkProvider)
     .build();
 ```
 
-The `JwkProvider` returned from the `JwkProviderBuilder` it's cached and rate limited, check it's [repository](https://github.com/auth0/jwks-rsa-java) to learn how to customize it.
+To use the **[Hybrid Flow](https://auth0.com/docs/api-auth/grant/hybrid)**, specify `id_token code` as the response type:
```

Comment:
```suggestion
To use the **[Hybrid Flow](https://auth0.com/docs/api-auth/grant/hybrid)**, specify `code id_token` as the response type:
```

Teeny tiny but if we ever do discovery, this would need to match [`response_types_supported `](https://auth0.auth0.com/.well-known/openid-configuration).

Reply:
https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#Terminology

**Multiple-Valued Response Types**
>The OAuth 2.0 specification allows for registration of space-separated response_type parameter values. If a Response Type contains one of more space characters (%20), it is compared as a space-delimited list of values in which the order of values does not matter.

Reply:
@lbalmaceda - Thank you for that ref! Makes more sense that way. Note to self to correct that in another SDK! :)